Apple builds fail sporadically

Builds on macOS sporadically fail with variants of the following error message:

```
error[E0463]: can't find crate for `serde_derive` which `serde` depends on
 --> components/data-provider/src/structs/decimal.rs:2:5
  |
2 | use serde::{Deserialize, Serialize};
  |     ^^^^^ can't find crate

error: aborting due to previous error
```

Sometimes this happens in the locale crate, and sometimes in data-provider.

Example: https://github.com/unicode-org/icu4x/pull/198/checks?check_run_id=957698689

@echeran

--- a/.github/workflows/build-test.yml
+++ b/.github/workflows/build-test.yml
@@ -34,24 +34,24 @@ jobs:
       with:
         path: ~/.cargo/registry
         key: ${{ runner.os }}-cargo-registry-${{ hashFiles('**/Cargo.toml') }}
-      # # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
-      # if: runner.os != 'macOS'
+      # TODO(actions-rs/cargo#111): make caching/WF config consistent across all runner OSes
+      if: runner.os != 'macOS'
 
     - name: Cache cargo index
       uses: actions/cache@v1
       with:
         path: ~/.cargo/git
         key: ${{ runner.os }}-cargo-index-${{ hashFiles('**/Cargo.toml') }}
-      # # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
-      # if: runner.os != 'macOS'
+      # TODO(actions-rs/cargo#111): make caching/WF config consistent across all runner OSes
+      if: runner.os != 'macOS'
 
     - name: Cache cargo build
       uses: actions/cache@v1
       with:
         path: target
         key: ${{ runner.os }}-cargo-build-target-${{ hashFiles('**/Cargo.toml') }}
-      # # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
-      # if: runner.os != 'macOS'
+      # TODO(actions-rs/cargo#111): make caching/WF config consistent across all runner OSes
+      if: runner.os != 'macOS'
 
     # Build job > Build and run steps
         

--- a/.github/workflows/build-test.yml
+++ b/.github/workflows/build-test.yml
@@ -34,32 +34,38 @@ jobs:
       with:
         path: ~/.cargo/registry
         key: ${{ runner.os }}-cargo-registry-${{ hashFiles('**/Cargo.toml') }}
-      # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
-      if: runner.os != 'macOS'
+      # # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
+      # if: runner.os != 'macOS'
 
     - name: Cache cargo index
       uses: actions/cache@v1
       with:
         path: ~/.cargo/git
         key: ${{ runner.os }}-cargo-index-${{ hashFiles('**/Cargo.toml') }}
-      # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
-      if: runner.os != 'macOS'
+      # # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
+      # if: runner.os != 'macOS'
 
     - name: Cache cargo build
       uses: actions/cache@v1
       with:
         path: target
         key: ${{ runner.os }}-cargo-build-target-${{ hashFiles('**/Cargo.toml') }}
-      # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
-      if: runner.os != 'macOS'
+      # # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
+      # if: runner.os != 'macOS'
 
     # Build job > Build and run steps
         
     - name: Build
-      run: cargo build ${{ matrix.feature-set }} --verbose
+      uses: actions-rs/cargo@v1.0.1
+      with:
+        command: build
+        args: ${{ matrix.feature-set }} --verbose
 
     - name: Test
-      run: cargo test ${{ matrix.feature-set }} --verbose
+      uses: actions-rs/cargo@v1.0.1
+      with:
+        command: test
+        args: ${{ matrix.feature-set }} --verbose
 
   # Formatting job
 

--- a/.github/workflows/build-test.yml
+++ b/.github/workflows/build-test.yml
@@ -34,18 +34,24 @@ jobs:
       with:
         path: ~/.cargo/registry
         key: ${{ runner.os }}-cargo-registry-${{ hashFiles('**/Cargo.toml') }}
+      # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
+      if: runner.os != 'macOS'
 
     - name: Cache cargo index
       uses: actions/cache@v1
       with:
         path: ~/.cargo/git
         key: ${{ runner.os }}-cargo-index-${{ hashFiles('**/Cargo.toml') }}
+      # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
+      if: runner.os != 'macOS'
 
     - name: Cache cargo build
       uses: actions/cache@v1
       with:
         path: target
         key: ${{ runner.os }}-cargo-build-target-${{ hashFiles('**/Cargo.toml') }}
+      # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
+      if: runner.os != 'macOS'
 
     # Build job > Build and run steps
         

--- a/.github/workflows/build-test.yml
+++ b/.github/workflows/build-test.yml
@@ -29,29 +29,29 @@ jobs:
 
     # Build job > Cache steps
       
-    # - name: Cache cargo registry
-    #   uses: actions/cache@v1
-    #   with:
-    #     path: ~/.cargo/registry
-    #     key: ${{ runner.os }}-cargo-registry-${{ hashFiles('**/Cargo.toml') }}
-    #   # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
-    #   if: runner.os != 'macOS'
+    - name: Cache cargo registry
+      uses: actions/cache@v1
+      with:
+        path: ~/.cargo/registry
+        key: ${{ runner.os }}-cargo-registry-${{ hashFiles('**/Cargo.toml') }}
+      # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
+      if: runner.os != 'macOS'
 
-    # - name: Cache cargo index
-    #   uses: actions/cache@v1
-    #   with:
-    #     path: ~/.cargo/git
-    #     key: ${{ runner.os }}-cargo-index-${{ hashFiles('**/Cargo.toml') }}
-    #   # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
-    #   if: runner.os != 'macOS'
+    - name: Cache cargo index
+      uses: actions/cache@v1
+      with:
+        path: ~/.cargo/git
+        key: ${{ runner.os }}-cargo-index-${{ hashFiles('**/Cargo.toml') }}
+      # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
+      if: runner.os != 'macOS'
 
     - name: Cache cargo build
       uses: actions/cache@v1
       with:
         path: target
         key: ${{ runner.os }}-cargo-build-target-${{ hashFiles('**/Cargo.toml') }}
-      ## Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
-      #if: runner.os != 'macOS'
+      # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
+      if: runner.os != 'macOS'
 
     # Build job > Build and run steps
         

--- a/.github/workflows/build-test.yml
+++ b/.github/workflows/build-test.yml
@@ -29,29 +29,29 @@ jobs:
 
     # Build job > Cache steps
       
-    - name: Cache cargo registry
-      uses: actions/cache@v1
-      with:
-        path: ~/.cargo/registry
-        key: ${{ runner.os }}-cargo-registry-${{ hashFiles('**/Cargo.toml') }}
-      # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
-      if: runner.os != 'macOS'
+    # - name: Cache cargo registry
+    #   uses: actions/cache@v1
+    #   with:
+    #     path: ~/.cargo/registry
+    #     key: ${{ runner.os }}-cargo-registry-${{ hashFiles('**/Cargo.toml') }}
+    #   # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
+    #   if: runner.os != 'macOS'
 
-    - name: Cache cargo index
-      uses: actions/cache@v1
-      with:
-        path: ~/.cargo/git
-        key: ${{ runner.os }}-cargo-index-${{ hashFiles('**/Cargo.toml') }}
-      # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
-      if: runner.os != 'macOS'
+    # - name: Cache cargo index
+    #   uses: actions/cache@v1
+    #   with:
+    #     path: ~/.cargo/git
+    #     key: ${{ runner.os }}-cargo-index-${{ hashFiles('**/Cargo.toml') }}
+    #   # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
+    #   if: runner.os != 'macOS'
 
     - name: Cache cargo build
       uses: actions/cache@v1
       with:
         path: target
         key: ${{ runner.os }}-cargo-build-target-${{ hashFiles('**/Cargo.toml') }}
-      # Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
-      if: runner.os != 'macOS'
+      ## Skip caching on macOS to prevent ephemeral build errors due to missing dependency files
+      #if: runner.os != 'macOS'
 
     # Build job > Build and run steps
         

